src/controllers/expenses.controller.js

```diff
@@ -0,0 +1,95 @@
+const { expensesService, usersService } = require('./../services');
+const { expenseSchema } = require('./../schemas');
+
+const getAll = (req, res) => {
+  const { userId, categories, from, to } = req.query;
+  const expenses = expensesService.getAll({
+    userId: +userId,
+    categories,
+    from,
+    to,
+  });
+
+  res.json(expenses);
+};
+
+const getOne = (req, res) => {
+  const { id } = req.params;
+
+  const expense = expensesService.getById(+id);
+
+  if (!expense) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.json(expense);
+};
+
+const create = (req, res) => {
+  const { error, value } = expenseSchema.validate(req.body, {
+    stripUnknown: true,
+  });
+
+  const isUserExist = !!usersService.getById(value.userId);
+
+  if (error || !isUserExist) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const newExpense = expensesService.create(value);
+
+  res.status(201).json(newExpense);
+};
+
+const removeOne = (req, res) => {
+  const { id } = req.params;
+
+  const isExist = !!expensesService.getById(+id);
+
+  if (!isExist) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  expensesService.deleteOne(+id);
+  res.sendStatus(204);
+};
+
+const update = (req, res) => {
+  const { id } = req.params;
+
+  const isExist = !!expensesService.getById(+id);
+
+  if (!isExist) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  const { error, value } = expenseSchema.validate(req.body, {
+    stripUnknown: true,
+  });
+
+  if (!error) {
+    res.sendStatus(400);
+
+    return;
+  }
```

Comment:
There is a logic error in the `update` function. The condition `if (!error)` should be `if (error)` to correctly handle validation errors. Currently, it sends a 400 status code when there is no error, which is incorrect.